services/user.py

```diff
@@ -0,0 +1,53 @@
+from django.db.models import QuerySet
+
+from db.models import User
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None
+) -> QuerySet:
+    user = User.objects.create_user(
+        username=username,
+        password=password,
+    )
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+    user.save()
+    return user
+
+
+def get_user(user_id: int) -> QuerySet:
+    return User.objects.get(id=user_id)
+
+
+def update_user(
+        user_id: int,
+        username: str = None,
+        password: str = None,
+        email: str = None,
+        first_name: str = None,
+        last_name: str = None
+
+) -> User:
+    user = User.objects.get(id=user_id)
```

Comment:
And here

Reply:
Thanks! You mean that.
If you reference [User] directly (for example, by referring to it in a foreign key), your code will not work in projects where the [AUTH_USER_MODEL] setting has been changed to a different user model.

get_user_model() you should reference the user model using django.contrib.auth.get_user_model(). This method will return the currently active user model – the custom user model if one is specified, or [User]nusing the [AUTH_USER_MODEL]

Reply:
As i understood from django doc we can use get_user_model() or author = models.ForeignKey(
        settings.AUTH_USER_MODEL,
        on_delete=models.CASCADE,
    )
 Generally speaking, it’s easiest to refer to the user model with the AUTH_USER_MODEL setting in code that’s executed at import time, however, it’s also possible to call get_user_model(). Thanks for proposition!

Reply:
use `get_user_model()`